src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -2015,7 +2015,6 @@ bool QgsPostgresProvider::addFeatures( QgsFeatureList &flist, Flags flags )
   {
     return false;
   }
-  conn->lock();
```

Comment:
Do we assume safety of the operations in here because they are serialized because any modifying operation needs to be run on the main thread?
I assume these create/update/delete blocks can coexist with reading values in between (in depth assessment of the statements sent required though), but among them they probably shouldn't run in parallel.
Not sure how safe it was before to run modifications on background threads (@nyalldawson @wonder-sk ?) before, but with this patch, a measure like this needs to be guaranteed.

Reply:
After a second read, it seems to me these locks serve mainly to make sure the database state is in sync with the provider. When updating the list of columns for instance, both the database and the provider should be modified in an atomic way.
So, yeah, better to keep them :) Thanks @m-kuhn